pydantic/_internal/_model_construction.py

```diff
@@ -202,6 +204,11 @@ def build_inner_schema(  # noqa: C901
                 raise
         raise PydanticUndefinedAnnotation(name) from e
 
+    # check for forward references and raise PydanticUndefinedAnnotation
+    for t in type_hints.values():
```

Comment:
This makes sense, but I don't think it needs to be a separate loop.

Reply:
Oops, I deleted my old reply/comment somehow. The point is that I think it does need to be a separate loop, since there is really no "other loop" to put this in (that I can find).

Reply:
line 214 we iterate over `type_hints`

```py
for ann_name, ann_type in type_hints.items():
```

Reply:
It was hiding in plain sight :)
Thanks!

Reply:
fixed and pushed